src/engine/engine.rs

```diff
@@ -74,6 +72,9 @@ pub trait Pool: HasName + HasUuid {
     /// Get the filesystem in this pool with this UUID.
     fn get_filesystem(&mut self, uuid: &FilesystemUuid) -> Option<&mut Filesystem>;
 
+    /// Does the pool have any created filesystems?
+    fn has_filesystems(&self) -> bool;
```

Comment:
You can leave this out of the interface, because the only place it is used is in a macro, and therefore your can leave out its implementations in the Pool implementation for StratPool and SimPool. I'm trying to restrict the Pool interface to things that the D-Bus API requires, and I think that's a good principle. You may need to put implementations of this method in the implementations for StratPool and SimPool.